tests/e2e/tests/checkout/blocks/pre-order-product.spec.js

```diff
@@ -0,0 +1,39 @@
+import { test, expect } from '@playwright/test';
+import config from 'config';
+import { api, payments, products } from '../../../utils';
+
+const { setupBlocksCheckout, fillCreditCardDetails } = payments;
+
+let productId;
+
+test.beforeAll( async () => {
+	productId = await api.create.product( products.preOrderData() );
+} );
+
+test.afterAll( async () => {
+	await api.deletePost.product( productId );
+} );
+
+test( 'customer can purchase a pre-order product @blocks @pre-orders', async ( {
+	page,
+} ) => {
+	await page.goto( `?p=${ productId }` );
```

Comment:
I guess we are using permalinks but this is easier for redirecting the browser to the product URL without knowing the post slug.